pkg/openstack/loadbalancer_test.go

```diff
@@ -593,3 +595,296 @@ func Test_getIntFromServiceAnnotation(t *testing.T) {
+							LBProvider: "ovn",
+							LBMethod:   "SOURCE_IP_PORT",
+						},
+					},
+				},
+				service: &corev1.Service{
+					Spec: corev1.ServiceSpec{
+						SessionAffinity: corev1.ServiceAffinityClientIP,
+					},
+				},
+			},
+			want: pools.CreateOpts{
+				Protocol:    pools.ProtocolHTTP,
+				LBMethod:    "SOURCE_IP_PORT",
+				Persistence: &pools.SessionPersistence{Type: "SOURCE_IP"},
+			},
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			got := tt.args.lbaasV2.buildPoolCreateOpt(tt.args.protocol, tt.args.service, tt.args.svcConf)
+			assert.Equal(t, got, tt.want)
+		})
+	}
+}
+
+func Test_getSubnetID(t *testing.T) {
+	type args struct {
+		svcConf *serviceConfig
+		service *corev1.Service
+		lbaasV2 *LbaasV2
+	}
+	tests := []struct {
+		name        string
+		args        args
+		want        string
+		expectedErr error
+	}{
+		{
+			name: "test get subnet from service annotation",
+			args: args{
+				svcConf: &serviceConfig{},
+				lbaasV2: &LbaasV2{
+					LoadBalancer{
+						opts: LoadBalancerOpts{
+							LBClasses: map[string]*LBClass{
+								"test-class": {
+									SubnetID:           "test-id",
+									FloatingNetworkID:  "test-floating-network-id",
+									FloatingSubnetID:   "test-floating-subnet-id",
+									FloatingSubnet:     "test-floating-subnet",
+									NetworkID:          "test-network-id",
+									FloatingSubnetTags: "test-floating-subnet-tags",
+									MemberSubnetID:     "test-member-subnet-id",
+								},
+							},
+						},
+					},
+				},
+				service: &corev1.Service{
+					ObjectMeta: v1.ObjectMeta{
+						Annotations: map[string]string{
+							"loadbalancer.openstack.org/subnet-id": "test-id",
+							"loadbalancer.openstack.org/class":     "test-class",
+						},
+					},
+				},
+			},
+			want: "test-id",
+		},
+		{
+			name: "test get subnet from config class",
+			args: args{
+				svcConf: &serviceConfig{},
+				lbaasV2: &LbaasV2{
+					LoadBalancer{
+						opts: LoadBalancerOpts{
+							LBClasses: map[string]*LBClass{
+								"test-class": {
+									SubnetID:           "test-class-subnet-id",
+									FloatingNetworkID:  "test-floating-network-id",
+									FloatingSubnetID:   "test-floating-subnet-id",
+									FloatingSubnet:     "test-floating-subnet",
+									NetworkID:          "test-network-id",
+									FloatingSubnetTags: "test-floating-subnet-tags",
+									MemberSubnetID:     "test-member-subnet-id",
+								},
+							},
+						},
+					},
+				},
+				service: &corev1.Service{
+					ObjectMeta: v1.ObjectMeta{
+						Annotations: map[string]string{
+							"loadbalancer.openstack.org/class": "test-class",
+						},
+					},
+				},
+			},
+			want:        "test-class-subnet-id",
+			expectedErr: nil,
+		},
+		{
+			name: "test get subnet from config class with invalid loadbalancer class",
+			args: args{
+				svcConf: &serviceConfig{},
+				lbaasV2: &LbaasV2{
+					LoadBalancer{
+						opts: LoadBalancerOpts{
```

Comment:
What do you think about adding one LBClass here that is NOT `test-class`?

Reply:
That sounds great, should I add an extra LBClass or completely replace the other LBClass?

Reply:
i've added an extra LBClass, if i completely remove it the function will set LBClass to "" and pick the default class instead